includes/Core/Authentication/Google_Proxy.php

```diff
@@ -118,30 +118,33 @@ public function get_user_fields() {
 	 * @since 1.5.0
 	 *
 	 * @param Credentials $credentials Credentials instance.
+	 * @param string      $mode        Sync mode.
+	 * @return array Response of the wp_remote_post request or NULL if there are no credentials.
 	 */
-	public function sync_site_fields( Credentials $credentials ) {
+	public function sync_site_fields( Credentials $credentials, $mode = 'async' ) {
 		if ( ! $credentials->has() ) {
-			return;
+			return null;
 		}
 
 		$creds = $credentials->get();
 
-		wp_remote_post(
-			$this->url( self::OAUTH2_SITE_URI ),
-			array(
-				'body'     => array_merge(
-					$this->get_site_fields(),
-					array(
-						'nonce'       => wp_create_nonce( self::ACTION_SETUP ),
-						'site_id'     => $creds['oauth2_client_id'],
-						'site_secret' => $creds['oauth2_client_secret'],
-					)
-				),
-				/** Don't block the process from finishing waiting for a response. @see \spawn_cron(). */
-				'timeout'  => 0.01,
-				'blocking' => false,
-			)
+		$request_args = array(
+			'body' => array_merge(
+				$this->get_site_fields(),
+				array(
+					'nonce'       => wp_create_nonce( self::ACTION_SETUP ),
+					'site_id'     => $creds['oauth2_client_id'],
+					'site_secret' => $creds['oauth2_client_secret'],
+				)
+			),
 		);
+
+		if ( 'sync' !== $mode ) {
```

Comment:
Not a big deal but let's be a bit more specific for now.
```suggestion
		if ( 'async' === $mode ) {
```